client.go

```diff
@@ -1081,17 +1099,20 @@ Loop:
 		switch {
 		// enable outgoing transfers case if sender and credits are available
 		case isSender && l.linkCredit > 0:
+			debug(1, "Link Mux isSender: credit: %d, deliveryCount: %d, messages: %d, pending: %d", l.linkCredit, l.deliveryCount, len(l.messages), len(l.pendingMessages))
 			outgoingTransfers = l.transfers
 
 		// if receiver && half maxCredits have been processed, send more credits
-		case isReceiver && l.linkCredit+uint32(len(l.messages)) <= l.receiver.maxCredit/2:
+		case isReceiver && l.linkCredit+uint32(len(l.pendingMessages)) <= (l.receiver.maxCredit)/2:
```

Comment:
The decision to emit a Flow frame is now based on `len(pendingMessages)` instead of `len(messages)` resulting in accurate flow control